dataland-frontend/src/components/forms/CreateEUTaxonomyForFinancials.vue

```diff
@@ -172,21 +507,167 @@ export default defineComponent({
+          await this.$nextTick();
+          const formInputsModelToSend = modifyObjectKeys(
+            JSON.parse(JSON.stringify(this.formInputsModel)) as objectType,
+            "send"
           );
-        this.$formkit.reset("createEuTaxonomyForFinancialsForm");
+          this.postEuTaxonomyDataForFinancialsResponse =
+            await euTaxonomyDataForFinancialsControllerApi.postCompanyAssociatedEuTaxonomyDataForFinancials(
+              formInputsModelToSend
+            );
+        }
       } catch (error) {
         this.postEuTaxonomyDataForFinancialsResponse = null;
         console.error(error);
       } finally {
         this.postEuTaxonomyDataForFinancialsProcessed = true;
+        this.confirmedSelectedKPIs = [];
+        this.selectedKPIs = [];
+        this.fiscalYearEnd = "";
+        this.filesToUpload = [];
+        this.uploadFiles = [];
+        this.formInputsModel = {};
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+        this.$refs.UploadReports.clearAllNotUploadedFiles();
+        await this.$nextTick();
+        this.$formkit.reset("createEuTaxonomyForFinancialsForm");
+      }
+    },
+
+    /**
+     * Add files to object filesToUpload
+     *
+     * @param event full event object containing the files
+     * @param event.originalEvent event information
+     * @param event.files files
+     */
+    onSelectedFilesHandler(event: { files: Record<string, string>[]; originalEvent: Event }): void {
+      if (event.files.length) {
+        event.files.forEach((file) => {
+          if (this.uploadFiles.some((objfile) => objfile.name === file.name.split(".")[0])) {
+            file["nameAlreadyExists"] = "true";
+          } else {
+            file["reportDate"] = file["reportDate"] ?? "";
+            file["convertedReportDate"] = file["convertedReportDate"] ?? "";
+            file["documentId"] = file["documentId"] ?? "";
+          }
+        });
+
+        this.filesToUpload = [...event.files] as ExtendedFile[];
+      } else {
+        return;
+      }
+    },
+
+    /**
+     * Remove report from files uploaded
+     *
+     * @param fileToRemove File To Remove
+     * @param fileRemoveCallback Callback function removes report from the ones selected in formKit
+     * @param index Index number of the report
+     */
+    removeReportFromFilesToUpload(
+      fileToRemove: Record<string, string>,
+      fileRemoveCallback: (x: number) => void,
+      index: number
+    ) {
+      fileRemoveCallback(index);
+      this.filesToUpload = this.filesToUpload.filter((el) => {
+        return el.name !== fileToRemove.name;
+      });
+    },
+
+    /**
+     * Update property in uploaded files
+     *
+     * @param indexFileToUpload Index number of the report
+     * @param property Property which is to be updated
+     * @param value Value to which it is to be changed
+     * @param whichSetOfFiles which set of files will be edited
+     */
+    updatePropertyFilesUploaded(
```

Comment:
Method is used also in non-financials Eu Taxo component, so maybe it should be put in a .ts utils class for EuTaxonomy instead of this component here.

Reply:
Yes. Right